src/languageclient.rs

```diff
@@ -256,6 +264,80 @@ impl State {
         Ok(())
     }
 
+    pub fn textDocument_documentHighlight(&mut self, params: &Option<Params>) -> Result<Value> {
+        self.textDocument_didChange(params)?;
+        info!("Begin {}", lsp::request::DocumentHighlightRequest::METHOD);
+        let (languageId, filename, line, character, handle): (
+            String,
+            String,
+            u64,
+            u64,
+            bool,
+        ) = self.gather_args(
+            &[
+                VimVar::LanguageId,
+                VimVar::Filename,
+                VimVar::Line,
+                VimVar::Character,
+                VimVar::Handle,
+            ],
+            params,
+        )?;
+
+        let result = self.call(
+            Some(&languageId),
+            lsp::request::DocumentHighlightRequest::METHOD,
+            TextDocumentPositionParams {
+                text_document: TextDocumentIdentifier {
+                    uri: filename.to_url()?,
+                },
+                position: Position { line, character },
+            },
+        )?;
+
+        if !handle {
+            return Ok(result);
+        }
+
+        let document_highlight: Option<Vec<DocumentHighlight>> =
+            serde_json::from_value(result.clone())?;
+        if let Some(document_highlight) = document_highlight {
+            let highlights = document_highlight
+                .into_iter()
+                .map(|DocumentHighlight { range, kind }| Highlight {
+                    line: range.start.line,
+                    character_start: range.start.character,
+                    character_end: range.end.character,
+                    group: self.documentHighlightDisplay[&kind
```

Comment:
Please change from index to `get()`, of which the former is hard crash in case index is not found.